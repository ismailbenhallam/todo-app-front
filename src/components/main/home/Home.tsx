import styled from "styled-components";
import NewTodoContainer from "./NewTodoContainer";
import { TodoPriority, TodoProps, TodoState } from "./Todo";
import TodoList from "./TodoList";

const Container = styled.div``;

const TODOS: Array<TodoProps> = [
  {
    id: 1,
    priority: TodoPriority.HIGH,
    title: "ETDDev, je veux initialiser le projet à l'aide de create-react-app",
    state: TodoState.WAITING,
    description:
      "text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text",
  },
  {
    id: 2,
    priority: TodoPriority.NORMAL,
    title: "ETDDev, je veux initialiser le projet à l'aide de create-react-app",
    state: TodoState.WAITING,
    description:
      "text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text",
  },
  {
    id: 3,
    priority: TodoPriority.LOW,
    title: "ETDDev, je veux initialiser le projet à l'aide de create-react-app",
    state: TodoState.WAITING,
    description:
      "text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text text",
  },
];

const Home = () => {
  return (
    <Container>
      <NewTodoContainer />
      <TodoList todos={TODOS} />
    </Container>
  );
};

export default Home;
