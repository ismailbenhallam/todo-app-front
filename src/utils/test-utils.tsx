import { render, RenderOptions } from "@testing-library/react";
import { FC, ReactElement } from "react";
import { Provider } from "react-redux";
import store from "redux/store";

export const AllProviders: FC = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
