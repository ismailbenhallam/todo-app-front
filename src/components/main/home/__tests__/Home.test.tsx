import renderer from "react-test-renderer";
import { AllProviders } from "../../../../utils/test-utils";
import Home from "../Home";

test("Home should match snapshot", () => {
  const tree = renderer
    .create(
      <AllProviders>
        <Home />
      </AllProviders>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
