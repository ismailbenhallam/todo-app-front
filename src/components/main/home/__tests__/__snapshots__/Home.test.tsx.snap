// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div>
  <form
    className="sc-bdfBQB hyTapZ"
    data-testid="form"
    onSubmit={[Function]}
  >
    <input
      className="sc-gsTEea cTQgfU"
      data-testid="title"
      name="title"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Titre"
    />
    <textarea
      className="sc-gsTEea sc-dlfnuX cTQgfU bLBrQh"
      data-testid="description"
      name="description"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Description"
    />
    <div
      className="sc-gKseQn bwLGZl"
      data-testid="error"
    >
      Veuillez saisir un titre SVP
    </div>
    <div
      className="sc-jSgvzq gyKoCI"
    >
      <select
        className="sc-hKgJUU gChHye"
        name="priority"
        onBlur={[Function]}
        onChange={[Function]}
      >
        <option
          value={0}
        >
          Importante
        </option>
        <option
          value={1}
        >
          Normale
        </option>
        <option
          value={2}
        >
          Faible
        </option>
      </select>
      <input
        className="sc-hKgJUU sc-eCstlR gChHye czdRiz"
        data-testid="button"
        type="submit"
        value="Ajouter"
      />
    </div>
  </form>
  <div
    className="sc-kstqJO cWaaAs"
    data-testid="TodoListContainer"
  >
    No data
  </div>
</div>
`;
